pandas/core/arrays/timedeltas.py

```diff
@@ -324,12 +325,105 @@ def _evaluate_with_timedelta_like(self, other, op):
 
     __mul__ = _wrap_tdi_op(operator.mul)
     __rmul__ = __mul__
-    __truediv__ = _wrap_tdi_op(operator.truediv)
     __floordiv__ = _wrap_tdi_op(operator.floordiv)
     __rfloordiv__ = _wrap_tdi_op(ops.rfloordiv)
 
+    def __truediv__(self, other):
+        other = lib.item_from_zerodim(other)
+
+        if isinstance(other, (ABCSeries, ABCDataFrame, ABCIndexClass)):
+            return NotImplemented
+
+        if isinstance(other, (timedelta, np.timedelta64, Tick)):
+            other = Timedelta(other)
+            if other is NaT:
+                # specifically timedelta64-NaT
+                result = np.empty(self.shape, dtype=np.float64)
+                result.fill(np.nan)
+                return result
+
+            # otherwise, dispatch to Timedelta implementation
+            return self._data / other
+
+        elif lib.is_scalar(other):
+            # assume it is numeric
+            result = self._data / other
+            freq = None
+            if self.freq is not None:
+                # Tick division is not implemented, so operate on Timedelta
+                freq = self.freq.delta / other
+            return type(self)(result, freq=freq)
+
+        if not hasattr(other, "dtype"):
+            # e.g. list, tuple
+            other = np.array(other)
+
+        if len(other) != len(self):
+            raise ValueError("Cannot divide vectors with unequal lengths")
+
+        elif is_timedelta64_dtype(other):
+            # let numpy handle it
+            return self._data / other
+
+        elif is_object_dtype(other):
+            result = [self[n] / other[n] for n in range(len(self))]
+            result = np.array(result)
```

Comment:
actually this is really close to what soft_convert_objects does.

Reply:
That isn't clear to me.  `soft_convert_objects` doesn't call `lib.infer_dtype` or any analogue.

Reply:
you are essentially re-implementing it. i would rather not do that.

Reply:
yes

Reply:
That's just swapping out `result, _ = sequence_to_td64ns(result)` with `result = soft_convert_objects(result, datetime=False, numeric=False, timedelta=True, coerce=False, copy=False)`, which in turn can be boiled down to a call to `result = lib.maybe_convert_objects(result, convert_timedelta=True)`, which involves a ton of unnecessary checks given that we already know that `lib.infer_dtype(result) == 'timedelta'`.  `sequence_to_td64ns` is the function for this use case.

Reply:
then before we do anything else,  pls use soft_convert_objects. you basically wrote all new routines. I actually don't have a problem with changing the usage of soft_convert_objects to use your new ones, but pls dont' write new things.

Reply:
> I actually don't have a problem with changing the usage of soft_convert_objects to use your new ones

`soft_convert_objects` is for a different use case.  Having that use `sequence_to_td64ns` would not make sense either.

> you basically wrote all new routines [...] but pls dont' write new things.

The "all new routine" you're talking about is literally one line in this method `result, _ = sequence_to_td64ns(result)` (`sequence_to_td64ns` is _not_ part of this PR).  I'm really not clear on a) why you think soft_convert_objects is a good fit here, and even more b) why using this specifically-designed-for-TimedeltaArray-constructor function bothers you so much.

Reply:
because we are constantly adding new code w/o using the existing code; its like the existing code is being completely bypassed and a new infrastructure is being written around it. i would much rather have the existing code be modified where need be and used.not using existing code just causes many many more issues later and re-discovered bugs.

Reply:
OK, I can get behind that goal.  Making a dedicated pass to see what parts of core.dtypes can be used to de-duplicate code is something I'd definitely be up for.

But the fact remains that soft_convert_objects is not appropriate here.

Reply:
@jreback updated to avoid the need for this discussion at all, and to allow for short-circuiting evaluation in non-timedelta cases.

Reply:
Another option would simply be to do no inference at all, just leave the result as object. That is eg what a numeric Series does with object array.

Reply:
ok reversing here. i agree with @jorisvandenbossche we don't infer object dtypes in ops (only in the constructor), so this seems reasonable

```
In [2]: pd.Series([1,2, 3]) / pd.Series([1,1,1], dtype=object)
Out[2]: 
0    1
1    2
2    3
dtype: object
```

Reply:
Fair enough, will change

Reply:
changed.  this ends up changing the behavior of the `DataFrame` test case, but that's largely driven by the fact that `DataFrame([NaT])` gets inferred as `datetime64[ns]`

Reply:
what is changed here? shouldn't is_object_type result in a TypeError or a NotImplemented?

Reply:
There are two independent questions that have been asked about the object-dtype case:

1) should we just raise TypeError instead or should we handle it so it Just Works (the latter being what this PR does)
2) Given that we handle this case, do we try to infer the output dtpye or just return object dtype?  This PR originally did the former, then changed to do the latter following discussion.

Reply:
ok this is fine, you are returning object dtype (which is consistent with how we do for Series now)